### PR TITLE
feat: support --update-goldens 

### LINKED ISSUE #8: Support --update-goldens

### PR DESCRIPTION
**Pull request by trejdych**
hello,
I have added support for --update-golden option. It works for --watch mode too, however, I had to filter some files from an observer.

I have added 'g' keyboard shortcut to toggling update goldens mode. 
CleanShot 2022-10-01 at 12 32 10

temporary I've pinned riverpod to 2.0.0-dev.3. 

Closes #8 

PS. It's my first contribution ever so sorry if I did something wrong.

### CODE REVIEW

File: packages/spec_cli/lib/src/dart_test.freezed.dart [hunk 1]
```diff
@@ -12,7 +12,7 @@ part of 'dart_test.dart';
 T _$identity<T>(T value) => value;
 
 final _privateConstructorUsedError = UnsupportedError(
-    'It seems like you constructed your class using `MyClass._()`. This constructor is only meant to be used by freezed and you are not supposed to need it nor use it.\nPlease check the documentation here for more information: https://github.com/rrousselGit/freezed#custom-getters-and-methods');
+    'It seems like you constructed your class using `MyClass._()`. This constructor is only meant to be used by freezed and you are not supposed to need it nor use it.\nPlease check the documentation here for more informations: https://github.com/rrousselGit/freezed#custom-getters-and-methods');
```
Comment (rrousselGit) [inline packages/spec_cli/lib/src/dart_test.freezed.dart:15]:
```suggestion
    'It seems like you constructed your class using `MyClass._()`. This constructor is only meant to be used by freezed and you are not supposed to need it nor use it.\nPlease check the documentation here for more information: https://github.com/rrousselGit/freezed#custom-getters-and-methods');
```

File: packages/spec_cli/lib/src/tests.freezed.dart [hunk 2]
```diff
@@ -12,7 +12,7 @@ part of 'tests.dart';
 T _$identity<T>(T value) => value;
 
 final _privateConstructorUsedError = UnsupportedError(
-    'It seems like you constructed your class using `MyClass._()`. This constructor is only meant to be used by freezed and you are not supposed to need it nor use it.\nPlease check the documentation here for more information: https://github.com/rrousselGit/freezed#custom-getters-and-methods');
+    'It seems like you constructed your class using `MyClass._()`. This constructor is only meant to be used by freezed and you are not supposed to need it nor use it.\nPlease check the documentation here for more informations: https://github.com/rrousselGit/freezed#custom-getters-and-methods');
```
Comment (rrousselGit) [inline packages/spec_cli/lib/src/tests.freezed.dart:15]:
```suggestion
    'It seems like you constructed your class using `MyClass._()`. This constructor is only meant to be used by freezed and you are not supposed to need it nor use it.\nPlease check the documentation here for more information: https://github.com/rrousselGit/freezed#custom-getters-and-methods');
```

File: packages/spec_cli/lib/src/io.dart [hunk 3]
```diff
@@ -15,7 +15,8 @@ final $fileChange = StreamProvider<void>((ref) {
 
   final dir = ref.watch($workingDirectory);
   final libChange = Directory(join(dir.path, 'lib')).watch(recursive: true);
-  final testChange = Directory(join(dir.path, 'test')).watch(recursive: true);
+  final testChange =
```
Comment (rrousselGit) [inline packages/spec_cli/lib/src/io.dart:18]: What's the reasoning for restarting tests when golden changes?

**Comment by rrousselGit** [comment]
It'd be great if you could write a test that uses `--update-goldens`

File: packages/spec_cli/lib/src/io.dart (hunk 3 above)
Comment (trejdych) [reply packages/spec_cli/lib/src/io.dart:18]: It's opposite. So probably the name is wrong :D 
It removes golden files from observing so test won't restart on the golden file changes (unless you move or remove them). this protects against an endless loop.

Comment (rrousselGit) [reply packages/spec_cli/lib/src/io.dart:18]: Oh my bad. Sounds good then

**Review by rrousselGit — APPROVED** [review]

**Comment by trejdych** [comment]
Sure! I'll try to add something.

**Comment by rrousselGit** [comment]
Awesome. I'll wait for it then. Otherwise your PR looks great, thanks for your work!

**Comment by trejdych** [comment]
it's my pleasure

**Comment by trejdych** [comment]
@rrousselGit I've added one test. Let me know if it is enough. 

if you don't mind, I would like to add tests for the `--watch` too (I could do it not only for --update-goldens). However I think it's much more complex topic. Have you thought about it?

File: packages/spec_cli/test/spec_cli_test.dart [hunk 4]
```diff
@@ -1062,4 +1062,53 @@ $testResults
...
+ PASS  packages/a/test/my_test.dart
+
+Test Suites: 1 passed, 1 total
+Tests:       1 passed, 1 total
+Time:        00:00:00
+''',
+    );
+    expect(
+        File('${workingDir.path}/packages/a/test/container.png').existsSync(),
+        true);
```
Comment (rrousselGit) [inline packages/spec_cli/test/spec_cli_test.dart:1111]: nit: trailing comma

**Comment by rrousselGit** [comment]
It's looking good

Watch mode is a bit tricky indeed. I haven't thought about it too much. I'm open to suggestions

File: packages/spec_cli/test/spec_cli_test.dart (hunk 4 above)
Comment (trejdych) [reply packages/spec_cli/test/spec_cli_test.dart]: done

**Comment by trejdych** [comment]
To be honest I don't have idea yet ;) but I’ll think about it

**Comment by rrousselGit** [comment]
Sorry for the delay. Thanks for this!

**Pull request merged**